assignment-1-sql/bdixon/brad_assignment_1.sql

```diff
@@ -0,0 +1,75 @@
+-- Assignment #1
+-- Question 1: Find the most recent completed order where the basket size was > $500.
+
+select id, user_id, completed_at, charge_amount
+from fact_order
+where charge_amount > 500
+and completed_at is not null
+order by completed_at desc
+limit 1
+
+-- Question 2: What GMV have we generated, how many orders were made per week in the 
+-- last 8 weeks in Canada? Order the results from the most recent week to the oldest.
+
+select date_trunc('week',completed_at) as completion_week, sum(f.charge_amount) as total_GMV, count(f.id) as order_count 
+from fact_order f
+    join zones z on f.zone_id = z.id 
+    join regions r on z.region_id = r.id
+where r.country_id = 124
+and completion_week between '2019-09-23' and '2019-11-11'
+group by 1
+order by 1 
+
+-- Question 3: What percentage of our active shoppers have been with us for the 
+-- following months: - (0-6) - (6-12) - (12-18) - (18-24) - 24+
+select case 
+    when datediff(month, hired_at, now()) between 0 and 6 then '0-6'
+    when datediff(month, hired_at, now()) between 7 and 12 then '7-12'
+    when datediff(month, hired_at, now()) between 13 and 18 then '12-18'
+    when datediff(month, hired_at, now()) between 19 and 24 then '18-24'
+    when datediff(month, hired_at, now()) > 24 then '>24'
+    END as shopper_age,
+    (count(id) / (select count(id) from drivers where hired_at is not null)) * 100 as shopper_percentage
+from drivers
```

Comment:
While this works, you might want to look into window functions (https://stackoverflow.com/questions/8515152/how-to-use-a-sql-window-function-to-calculate-a-percentage-of-an-aggregate) to keep from repeatedly counting the total number of drivers.